### ISSUE TITLE
Range functions should be exported

### ISSUE THREAD

**Issue by clauswilke** [tidyverse/ggplot2#2710]
I think the functions in https://github.com/tidyverse/ggplot2/blob/master/R/range.r should be exported. They are needed by anybody who wants to write their own scale functions from scratch.

Possibly they would fit better into the scales package, but since they depend on ggproto there's a dependency problem.

**Comment by hadley** [comment]
I think we could easily rewrite to remove the dependency issue.

**Comment by dpseidel** [comment]
Isn't this already done in scales: https://github.com/r-lib/scales/blob/master/R/range.r ?

**Comment by clauswilke** [comment]
Indeed, looks like that is the case already. Is there any reason why ggplot2 shouldn’t just use those?

**Comment by hadley** [comment]
I guess we forgot? 

@dpseidel do you want to do a PR?

**Comment by dpseidel** [comment]
Sure thing. I'll work one up this week.

**Comment by paleolimbot** [comment]
@dpseidel Are the range functions now safe to use for ggplot2?

**Comment by dpseidel** [comment]
The situation remains the same: the range functions are fixed in the dev version of scales but we have not had a new release. Now that my dissertation is (all but) finished, I've been meaning to give scales some more attention. Perhaps we can try for a release by end of summer and get this fixed.

**Comment by karawoo** [comment]
It looks like this should be ready now that scales 1.1.0 was released last fall, @dpseidel is that right?

**Comment by dseidel-plenty** [comment]
@karawoo  Should be! The updated range functions were released with scales 1.1.0

**Comment by dpseidel** [comment]
@karawoo Should be! The updated range functions were released with scales 1.1.0

**Comment by dpseidel** [comment]
@karawoo Tried to build a PR for this just now and am kicking myself because I've discovered `scales::DiscreteRange` does not currently support the `na.rm` argument to `train_discrete` which is critical. I will open a PR to scales now. 

In lieu of waiting for a new scales release to make this switch, we could still eliminate the ggproto dependencies now but we will need to include a custom DiscreteRange object with the correct train function. I've made and tested that change [here](https://github.com/dpseidel/ggplot2/blob/rangeimport2/R/range.r). 

Is it a priority to make this switch now? Or should we continue to wait for this to be completely patched in scales?
